packages/app/modules/auth/components/AuthWrapper.tsx

```diff
@@ -14,6 +16,60 @@ export const AuthWrapper = ({
   children,
   unauthorizedElement,
 }: AuthWrapperProps) => {
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const { currentTheme } = useTheme();
+
+  const authenticate = async () => {
+    const hasHardware = await LocalAuthentication.hasHardwareAsync();
+    if (!hasHardware) {
+      Alert.alert(
+        'Error',
+        'Your device does not support biometric authentication.',
+      );
+      return;
+    }
+
+    const hasBiometrics = await LocalAuthentication.isEnrolledAsync();
+    if (!hasBiometrics) {
+      Alert.alert(
+        'Error',
+        'No biometrics are enrolled. Please set up biometrics in your device settings.',
+      );
+      return;
+    }
+
+    const result = await LocalAuthentication.authenticateAsync({
+      promptMessage: 'Authenticate to continue',
+      fallbackLabel: 'Use Passcode',
+    });
+
+    if (result.success) {
+      setIsAuthenticated(true);
+    }
+  };
+
+  useEffect(() => {
+    authenticate();
+  }, []);
+
+  if (!isAuthenticated) {
```

Comment:
I believe this should be either platform specific or it be set on web as well like using WebAuthn or so. 

or just bypass it on web cause this won't allow pages wrapped with the context to be accessed.